### PR TITLE
Add describedBy words to contributors in bylines

### PR DESCRIPTION
**Pull request by davidpmccormick**
References #4479

Adding what a contributor _does_ to their mention in the byline.

image

Added the concept of `describedBy` to contributor roles in Prismic, with these two for starters (if this is incorrect, it is content editable in Prismic, of course).

| role | describedBy |
|-----------|----------|
|Author | words |
|Photographer | images |

### CODE REVIEW

File: prismic-model/json/editorial-contributor-roles.json [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+      "type": "StructuredText",
+      "config": {
+        "single": "paragraph,hyperlink,strong,em,heading1",
+        "label": "Title"
+      }
+    },
+    "describedBy": {
+      "type": "Text",
+      "config": {
+        "label": "Byline word for role",
```
Comment (jamesgorrie) [inline prismic-model/json/editorial-contributor-roles.json]: `Word describing the role [in the byline]`

Adding the byline ties it to design, which could change and makes editorial people feel like they're controlling design, where it's content that should portable. 

I am also being 💎 about it.

Comment (davidpmccormick) [reply prismic-model/json/editorial-contributor-roles.json]: Yeah, makes sense.

File: common/services/prismic/parsers.js [hunk 2]
```diff
@@ -250,6 +250,7 @@ export function parseContributors(
           ? {
               id: contributor.role.id,
               title: asText(contributor.role.data.title) || '',
+              describedBy: contributor.role.data.describedBy || '',
```
Comment (jamesgorrie) [inline common/services/prismic/parsers.js]: Is there a reason no to make this nullable?

File: common/utils/capitalise.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// @flow
+export default (word: string): string => {
+  const [firstLetter] = word;
+  const restOfWord = word.slice(1);
+
+  return `${firstLetter.toUpperCase()}${restOfWord}`;
+};
```
Comment (jamesgorrie) [inline common/utils/capitalise.js]: Is there not a grammar file we can put all of these into?
Don't really mind, just a thought.

File: content/webapp/pages/article.js [hunk 4]
```diff
@@ -142,10 +143,16 @@ export class ArticlePage extends Component<Props, State> {
                 [font('hnl', 6)]: true,
               })}
             >
-              {article.contributors.length > 0 && <span>By </span>}
-
-              {article.contributors.map(({ contributor }, i, arr) => (
+              {article.contributors.map(({ contributor, role }, i, arr) => (
                 <Fragment key={contributor.id}>
+                  {role && role.describedBy && (
```
Comment (jamesgorrie) [inline content/webapp/pages/article.js]: We check for null here, so it should probably be nullable above.

File: common/utils/capitalise.js (hunk 3 above)
Comment (davidpmccormick) [reply common/utils/capitalise.js]: yeah, I wondered the same – `grammar.js`?

Comment (davidpmccormick) [reply common/utils/capitalise.js]: Next question – `capitalise/camelise` or `capitalize/camelize`?

**Pull request merged**